crates/ra_assists/src/handlers/unwrap_block.rs

```diff
@@ -0,0 +1,412 @@
+                    foo();
+
+                    //comment
+                    bar();
+                } else {
+                    println!("bar");
+                }
+            }
+            "#,
+        );
+    }
+
+    #[test]
+    fn issue_example_with_if() {
+        check_assist(
+            unwrap_block,
+            r#"
+            fn complete_enum_variants(acc: &mut Completions, ctx: &CompletionContext, ty: &Type) {
+                if let Some(ty) = &ctx.expected_type {<|>
+                    if let Some(Adt::Enum(enum_data)) = ty.as_adt() {
+                        let variants = enum_data.variants(ctx.db);
+
+                        let module = if let Some(module) = ctx.scope().module() {
+                            // Compute path from the completion site if available.
+                            module
+                        } else {
+                            // Otherwise fall back to the enum's definition site.
+                            enum_data.module(ctx.db)
+                        };
+
+                        for variant in variants {
+                            if let Some(path) = module.find_use_path(ctx.db, ModuleDef::from(variant)) {
+                                // Variants with trivial paths are already added by the existing completion logic,
+                                // so we should avoid adding these twice
+                                if path.segments.len() > 1 {
+                                    acc.add_enum_variant(ctx, variant, Some(path.to_string()));
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            "#,
+            r#"
+            fn complete_enum_variants(acc: &mut Completions, ctx: &CompletionContext, ty: &Type) {
+                <|>if let Some(Adt::Enum(enum_data)) = ty.as_adt() {
+                    let variants = enum_data.variants(ctx.db);
+
+                    let module = if let Some(module) = ctx.scope().module() {
+                        // Compute path from the completion site if available.
+                        module
+                    } else {
+                        // Otherwise fall back to the enum's definition site.
+                        enum_data.module(ctx.db)
+                    };
+
+                    for variant in variants {
+                        if let Some(path) = module.find_use_path(ctx.db, ModuleDef::from(variant)) {
+                            // Variants with trivial paths are already added by the existing completion logic,
+                            // so we should avoid adding these twice
+                            if path.segments.len() > 1 {
+                                acc.add_enum_variant(ctx, variant, Some(path.to_string()));
+                            }
+                        }
+                    }
+                }
+            }
+            "#,
+        );
+    }
```

Comment:
It's better not to add such vebose tests, as later it's pretty hard to figure out what the test is about. Tests should be minimal.